internal/encoder/compiler.go

```diff
@@ -487,7 +487,7 @@ func (c *Compiler) listElemCode(typ *runtime.Type) (Code, error) {
 	case typ.Kind() == reflect.Map:
 		return c.ptrCode(runtime.PtrTo(typ))
 	default:
-		code, err := c.typeToCodeWithPtr(typ, false)
+		code, err := c.typeToCodeWithPtr(typ, true)
```

Comment:
Hmm..., I feel that the fix to set `isPtr` to `true` is not correct to fix this issue ( 
since elem is not ptr type )

Reply:
I added comment!